Userland/DevTools/Playground/main.cpp

```diff
@@ -231,6 +231,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         auto formatted_gml = GUI::format_gml(source);
         if (!formatted_gml.is_null()) {
             editor->set_text(formatted_gml);
+            window->set_modified(true);
```

Comment:
Maybe we should only do this, and the set_text above, if the `formatted_gml` is not equal to the `source`? Otherwise we will set the window as modified even if nothing has actually changed during formatting?

Reply:
@bgianfo that sounds good – happy to add that recommendation after I figure out inconsistency with the "modified" state. It seems like whether or not something is modified is closely coupled to the undo stack of the underlying document.

I'm trying to understand the best way to signify/invoke `on_modified_change` to ensure we have correct behavior. You can observe this by following the steps in the linked issue and trying to save (via the File menu, or with the shortcut), the document will be marked as "modified" but won't actually save. Am I missing a piece here? :^)